File: app/src/main/java/dev/leonlatsch/kolibri/boot/jobs/base/Job.java
package dev.leonlatsch.kolibri.boot.jobs.base;

import android.content.Context;

/**
 * Base class for sync jobs
 *
 * @author Leon Latsch
 * @since 1.0.0
 */
public abstract class Job extends BaseJob {

    protected Job(Context context) {
        super(context);
    }

    public abstract JobResult execute();
}

File: app/src/main/java/dev/leonlatsch/kolibri/boot/jobs/base/Job.kt
package dev.leonlatsch.kolibri.boot.jobs.base

import android.content.Context

/**
 * Base class for sync jobs
 *
 * @author Leon Latsch
 * @since 1.0.0
 */
abstract class Job protected constructor(context: Context) : BaseJob(context) {

    abstract fun execute(): JobResult
}
